pystac_client/item_search.py

```diff
@@ -164,9 +164,6 @@ class ItemSearch:
             will continue to request pages of items until the number of max items is
             reached. By default (``max_items=None``) all items matching the query
             will be returned.
-        stac_io: An instance of StacIO for retrieving results. Normally comes
-            from the Client that returns this ItemSearch client: An instance of a
-            root Client used to set the root on resulting Items.
```

Comment:
Same as `ignore_conformance`, I'd like to deprecate this argument, make it a noop, and point folks to the right way to do things.

Reply:
This one actually felt like an unnecessary change.